EIPS/Skinny_CREATE2.md

```diff
@@ -0,0 +1,16 @@
+```
+EIP: <to be assigned>
+Title: Skinny CREATE2
+Author: Vitalik Buterin
+Category: Core
+Status: Draft
+Created: 2018-04-20
+```
+
+### Specification
+
+Adds a new opcode at 0xf5, which takes 4 stack arguments: endowment, memory_start, memory_length, salt. Behaves identically to CREATE, except using `sha3(msg.sender ++ salt ++ init_code)[12:]` instead of the usual sender-and-nonce-hash as the address where the contract is initialized at.
```

Comment:
@vbuterin The salt can have any arbitrary length I guess?

Reply:
Existing scheme looks like this for low nonces:
```
sha3(0xd6 ++ 0x94 ++ sender ++ 0x01)
```
So if you mine an address starting with `d694`, it seems possible to create destination collisions. Using larger nonces give more room for collisions. 

This means that you could say "look, this contract can only be `create2`:ed with the initcode `x`. But in fact, you can create arbitrary contracts there using your vanity-mined `sender` and old-style `create`. 

I may be wrong, thinking while writing here... 

And @LefterisJP @vbuterin I assume the `salt` is fixed-size, and the size of that would naturally affect the ability to do the attack described here.

Reply:
> @vbuterin The salt can have any arbitrary length?

It's a stack argument, hence 32 bytes.

> A trivial way to get around this would be to prefix the entire thing with something that is invalid rlp

If we want, we can prefix with 0xff; the only valid RLP that starts with 0xff would be petaby long.